### PR TITLE
Feature: search project bar

### LINKED ISSUE #254: HomePage: Search bar in project grid
## Description

Implements search bar in project grid. 

Work like the git branch search bar.

Only filter on the project name

Expected design:
image

### PR DESCRIPTION
**Pull request by Zorin95670**
Resolved #254

### CODE REVIEW

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/composables/Filter.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Indicate if text contains searched text.
+ * @param {String} textToFilter - Text to filter.
+ * @param {String} textToSearch - Text to search.
+ * @return {Boolean} True if `textToFilter` contains `textToSearch` otherwise false.
+ */
+export function textFilter(textToFilter, textToSearch) {
```
Comment (k-y0u) [inline src/composables/Filter.js:7]: textFilter -> searchFilter

File: cypress/e2e/HomePage/filter.project.feature [hunk 2]
```diff
@@ -54,3 +55,33 @@ Feature: Test home page: project filter
...
+    # Verify all project tags are inactive
+    And  I expect '[data-cy="inactive-tag_local"]' exists
+    And  I expect '[data-cy="inactive-tag_remote"]' exists
+
+    # Set 'local' as searched text and expect only local projects are displayed
+    When I set on '[data-cy="search-project-input"]' text 'local'
+    And  I expect '[data-cy="project-card_{{remoteProjectName}}"]' not exists
+    And  I expect '[data-cy="project-card_{{localProjectName}}"]' appear 1 time on screen
+
+    # Set 'leto' as searched text and expect only remote projects are displayed
```
Comment (k-y0u) [inline cypress/e2e/HomePage/filter.project.feature:69]: cute :heart:

File: cypress/e2e/HomePage/filter.project.feature [hunk 3]
```diff
@@ -54,3 +55,33 @@ Feature: Test home page: project filter
...
+    When I set on '[data-cy="search-project-input"]' text 'leto'
+    And  I expect '[data-cy="project-card_{{localProjectName}}"]' not exists
+    And  I expect '[data-cy="project-card_{{remoteProjectName}}"]' appear 1 time on screen
+
+    # Set 'none' as searched text and expect no projects are displayed
```
Comment (k-y0u) [inline cypress/e2e/HomePage/filter.project.feature:74]: `no project is`, no ?

File: cypress/e2e/HomePage/filter.project.feature [hunk 4]
```diff
@@ -54,3 +55,33 @@ Feature: Test home page: project filter
     And  I expect '[data-cy="inactive-tag_local"]' exists
     And  I expect '[data-cy="project-card_{{remoteProjectName}}"]' appear 1 time on screen
     And  I expect '[data-cy="project-card_{{localProjectName}}"]' not exists
+
+  Scenario: Set/unset searched text should filter the projects list
+    # Verify all project tags are inactive
+    And  I expect '[data-cy="inactive-tag_local"]' exists
+    And  I expect '[data-cy="inactive-tag_remote"]' exists
+
+    # Set 'local' as searched text and expect only local projects are displayed
```
Comment (k-y0u) [inline cypress/e2e/HomePage/filter.project.feature:64]: bad description

File: cypress/e2e/HomePage/filter.project.feature (hunk 2 above)
Comment (k-y0u) [inline cypress/e2e/HomePage/filter.project.feature:69]: bad description

File: tests/unit/composables/Filter.spec.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+describe('Test composable: Filter', () => {
+  describe('Test function: textFilter', () => {
+    it('should return true without searched text', async () => {
+      expect(textFilter('abc', null)).toEqual(true);
+      expect(textFilter('abc', '')).toEqual(true);
+      expect(textFilter('abc', ' ')).toEqual(true);
+    });
+
+    it('should return true if searched text is contains', async () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/composables/Filter.spec.js:11]: done

File: tests/unit/composables/Filter.spec.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      expect(textFilter('abc', 'a')).toEqual(true);
+      expect(textFilter('abc', 'A')).toEqual(true);
+      expect(textFilter('abc', 'a b')).toEqual(true);
+    });
+
+    it('should return false if searched text is not contains', async () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/composables/Filter.spec.js:17]: done

File: tests/unit/components/grid/ProjectGrid.spec.js [hunk 7]
```diff
@@ -53,4 +61,16 @@ describe('Test component: ProjectGrid', () => {
       expect(wrapper.vm.tags[1].active).toEqual(false);
     });
   });
+
+  describe('Test watcher: props.projects', () => {
+    it('should be triggered when props.status is updated', async () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/components/grid/ProjectGrid.spec.js:66]: done

File: changelog.md [hunk 8]
```diff
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     * A short projects list access.
   * Sort projects by creating date from newest to oldest.
   * Add tags list to filter project.
+  * Add search bar for projects
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply changelog.md:18]: done

File: src/composables/Filter.js (hunk 1 above)
Comment (Zorin95670) [reply src/composables/Filter.js:7]: See with you, replace by `searchText`, done

File: src/components/grid/ProjectGrid.vue [hunk 9]
```diff
@@ -15,14 +15,25 @@
...
       <h5 class="text-primary q-ma-md">
         {{ $t('page.home.project.recent') }}
       </h5>
       <tag-list
         :tags="tags"
         @toggle:tag="onToggleTag"
       />
+      <q-input
+        v-model="searchProjectText"
+        outlined
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/grid/ProjectGrid.vue:28]: done

File: cypress/e2e/HomePage/filter.project.feature (hunk 3 above)
Comment (Zorin95670) [reply cypress/e2e/HomePage/filter.project.feature:74]: right, done

File: cypress/e2e/HomePage/filter.project.feature (hunk 4 above)
Comment (Zorin95670) [reply cypress/e2e/HomePage/filter.project.feature:64]: done

File: cypress/e2e/HomePage/filter.project.feature (hunk 2 above)
Comment (Zorin95670) [reply cypress/e2e/HomePage/filter.project.feature:69]: done

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/grid/ProjectGrid.vue [hunk 10]
```diff
@@ -15,14 +15,25 @@
...
+        class="search-bar"
+        :label="$t('actions.default.search')"
+        @update:model-value="updateProjects"
+      >
+        <template #prepend>
+          <q-icon name="fa-solid fa-magnifying-glass" />
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/grid/ProjectGrid.vue]: done

**Review by justinehell — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**